### PR TITLE
Improving the addition of invitations

### PR DESCRIPTION
**Pull request by arek1-618**
Improving the addition of invitations
## Edit
firefox_2019-03-12_08-54-06

## Detail
firefox_2019-03-12_11-56-34

### CODE REVIEW

File: install/install_schema/scheme.sql [hunk 1]
```diff
@@ -1112,6 +1112,7 @@ CREATE TABLE `u_yf_activity_invitation` (
   `activityid` int(10) NOT NULL,
   `crmid` int(10) NOT NULL DEFAULT 0,
   `email` varchar(100) NOT NULL DEFAULT '',
+  `name` varchar(255) DEFAULT NULL,
```
Comment (mariuszkrzaczkowski) [inline install/install_schema/scheme.sql]:
```suggestion
  `name` varchar(500) DEFAULT NULL,
```

File: layouts/basic/modules/Calendar/EditViewBlocks.tpl [hunk 2]
```diff
@@ -8,28 +8,39 @@
...
 			</div>
-			<div class="col-md-4 fieldRow">
-				<input type="text" title="{\App\Language::translate('LBL_SELECT_INVITE', $MODULE)}" placeholder="{\App\Language::translate('LBL_SELECT_INVITE', $MODULE)}" class="form-control form-control-sm inviteesSearch"/>
+			<div class="col-md-5 input-group js-popover-tooltip" data-js="popover" data-content="{\App\Language::translate('LBL_SELECT_INVITE', $MODULE_NAME)}">
+				<div class="input-group-prepend">
+					<span class="input-group-text u-cursor-pointer">
+						<span class="fas fa-search"></span>
+					</span>
+				</div>
+				<input type="text" class="form-control inviteesSearch" title="{\App\Language::translate('LBL_SELECT_INVITE', $MODULE_NAME)}"
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/EditViewBlocks.tpl]: inviteesSearch > js-participants-search

File: languages/en-US/Calendar.json [hunk 3]
```diff
@@ -161,8 +164,8 @@
     "LBL_SM_TUE": "Tue",
     "LBL_SM_WED": "Wed",
     "LBL_UNTIL": "Until",
-    "LBL_INVITE_RECORDS": "Invite users",
-    "LBL_SELECT_INVITE": "Invite others",
+    "LBL_INVITE_RECORDS": "Invite participants",
```
Comment (mariuszkrzaczkowski) [inline languages/en-US/Calendar.json]:
```suggestion
    "LBL_PARTICIPANTS_RECORDS": "Invite participants",
```

File: modules/Calendar/views/InviteEmail.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	 * {@inheritdoc}
+	 */
+	public $modalSize = 'modal-dialog-centered';
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function checkPermission(App\Request $request)
+	{
+		if (!\App\Privilege::isPermitted($request->getModule())) {
```
Comment (mariuszkrzaczkowski) [inline modules/Calendar/views/InviteEmail.php]: ?

File: public_html/layouts/basic/modules/Calendar/resources/Edit.js [hunk 5]
```diff
@@ -478,27 +529,31 @@ Vtiger_Edit_Js("Calendar_Edit_Js", {}, {
...
-					var inviteRow = inviteesContent.find('.d-none .inviteRow').clone(true, true);
+					let inviteRow = inviteesContent.find('.d-none .inviteRow').clone(true, true);
 					Vtiger_Index_Js.getEmailFromRecord(selected.id, selected.module).done(function (email) {
 						inviteRow.data('crmid', selected.id);
 						inviteRow.data('email', email);
 						inviteRow.find('.inviteName').data('content', selected.fullLabel + email).text(selected.label);
 						inviteRow.find('.inviteIcon .c-badge__icon').removeClass('fas fa-envelope').addClass('userIcon-' + selected.module);
 						inviteesContent.append(inviteRow);
 					});
+				}else{
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Calendar/resources/Edit.js]: ?

File: layouts/basic/modules/Calendar/Detail/BlockView.tpl [hunk 6]
```diff
@@ -7,8 +7,8 @@
 	<div class="detailViewTable">
 		<div class="js-toggle-panel c-panel" data-js="click" data-label="LBL_INVITE_RECORDS">
 			<div class="blockHeader c-panel__header">
-				<span class="js-block-toggle fas fa-angle-right m-2 {if !($IS_HIDDEN)}d-none{/if}" data-js="click" alt="{\App\Language::translate('LBL_EXPAND_BLOCK')}" data-mode="hide" data-id='INVITE_USER_BLOCK_ID'></span>
-				<span class="js-block-toggle fas fa-angle-down m-2 {if $IS_HIDDEN}d-none{/if}" data-js="click" alt="{\App\Language::translate('LBL_COLLAPSE_BLOCK')}" data-mode="show" data-id='INVITE_USER_BLOCK_ID'></span>
+				<span class="js-block-toggle fas fa-angle-right m-2 {if !($IS_HIDDEN)}d-none{/if}" data-js="click" alt="{\App\Language::translate('LBL_EXPAND_BLOCK')}" data-mode="hide" data-id='INVITE_PARTICIPANT_BLOCK_ID'></span>
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Calendar/Detail/BlockView.tpl]: inviteParticipantBlockId

File: modules/API/models/CalDAV.php [hunk 7]
```diff
@@ -524,19 +524,21 @@ protected function recordSaveAttendee(Vtiger_Record_Model $record, Sabre\VObject
...
 						'time' => $timeFormated,
+						'name' => \App\TextParser::textTruncate($nameAttendee, 500, false),
 					], ['activityid' => $record->getId(), 'email' => $value]
 					)->execute();
 				}
 				unset($invities[$value]);
 			} else {
 				$params = [
-					'email' => $value,
+					'email' => \App\TextParser::textTruncate($value, 100, false),
```
Comment (rskrzypczak) [inline modules/API/models/CalDAV.php]: ?

**Pull request merged**